r-package/grf/tests/testthat/test_causal_forest.R

```diff
@@ -155,32 +155,38 @@ test_that("predictions are invariant to scaling of the sample weights.", {
 })
 
 test_that("IPCC weighting in the training of a causal forest with missing data improves its complete-data MSE.", {
-  n <- 800
-  p <- 2
+  n <- 1000
+  p <- 6
   X <- matrix(rnorm(n * p), n, p)
-  e <- 1 / (1 + exp(-3 * X[, 1] + 2 * X[, 2]))
+  e <- 1 / (1 + exp(-X[, 1] + 2 * X[, 2]))
   W <- rbinom(n, 1, e)
-  tau <- 20 * X[, 1] - X[, 2]
+  tau <- 2 * (X[, 1] > 0 & X[,5] > 0) -
+      0.5 * (X[, 2] > 0) - 0.5 * (X[, 3] > 0) - 0.5 * (X[, 4] > 0)
   Y <- W * tau + rnorm(n)
 
-  e.cc <- 1 / (1 + exp(-2 * X[, 1]))
+  e.cc <- 1 - 0.9 * (X[, 1] > 0 & X[,5] > 0)
   cc <- as.logical(rbinom(n, 1, e.cc))
   sample.weights <- 1 / e.cc
 
-  num.trees <- 400
+  num.trees <- 500
+  mse <- function(f) {
+      tau.hat = rep(NA, n)
+      tau.hat[cc] = predict(f)$predictions
+      tau.hat[!cc] = predict(f, X[!cc,])$predictions
+      mean((tau.hat - tau)^2)
+  }
+  
   forest <- causal_forest(X[cc, ], Y[cc], W[cc], num.trees = num.trees)
-  boosted.forest <- causal_forest(X[cc, ], Y[cc], W[cc], orthog.boosting = TRUE, num.trees = num.trees)
   weighted.forest <- causal_forest(X[cc, ], Y[cc], W[cc], sample.weights = sample.weights[cc], num.trees = num.trees)
+  expect_true(mse(weighted.forest) / mse(forest) < .9)
+  
+  boosted.forest <- causal_forest(X[cc, ], Y[cc], W[cc], orthog.boosting = TRUE, num.trees = num.trees)
   boosted.weighted.forest <- causal_forest(
-    X[cc, ], Y[cc], W[cc],
-    sample.weights = sample.weights[cc],
-    orthog.boosting = TRUE, num.trees = num.trees
+      X[cc, ], Y[cc], W[cc],
+      sample.weights = sample.weights[cc],
+      orthog.boosting = TRUE, num.trees = num.trees
   )
-  mse <- function(f) {
-    mean((predict(f, X)$predictions - tau)^2)
-  }
-  expect_true(mse(weighted.forest) / mse(forest) < .975)
-  expect_true(mse(boosted.weighted.forest) / mse(boosted.forest) < .975)
+  expect_true(mse(boosted.weighted.forest) / mse(boosted.forest) < .9)
```

Comment:
I find it just a little more convenient to use `expect_lt` here instead of `expect_true`, because it lets the user know the difference. This is especially useful when one of the elements being compared happens to be NA (likely due to a bug), eg.
```r
> x <- NA  # a bug
> y <- 0
> expect_true(x < y)
# Error: x < y isn't true.
> expect_lt(x, y)
# Error: `x` is not strictly less than `y`. Difference: NA
```

Reply:
good point, thanks!